src/analysis/Environments/PauseButton.ts

```diff
@@ -1,33 +1,41 @@
+import _ from 'lodash/fp';
 import { ReactNode } from 'react';
 import { h } from 'react-hyperscript-helpers';
-import { getDisplayStatus, isComputePausable } from 'src/analysis/utils/resource-utils';
+import { getDisplayStatus } from 'src/analysis/utils/resource-utils';
 import { getToolLabelFromCloudEnv, isPauseSupported } from 'src/analysis/utils/tool-utils';
 import { Link } from 'src/components/common';
 import { makeMenuIcon } from 'src/components/PopupTrigger';
-import { App } from 'src/libs/ajax/leonardo/models/app-models';
-import { ListRuntimeItem } from 'src/libs/ajax/leonardo/models/runtime-models';
+import { App, isApp } from 'src/libs/ajax/leonardo/models/app-models';
+import {
+  isRuntime,
+  LeoRuntimeStatus,
+  ListRuntimeItem,
+  Runtime,
+  runtimeStatuses,
+} from 'src/libs/ajax/leonardo/models/runtime-models';
+import * as Utils from 'src/libs/utils';
 
 import { LeoResourcePermissionsProvider } from './Environments.models';
 
-type PausePermissionsProvider = Pick<LeoResourcePermissionsProvider, 'canPauseResource'>;
+type PausePermissionsProvider = Pick<LeoResourcePermissionsProvider, 'hasPausePermission'>;
 
-interface PauseButtonProps {
+export interface PauseButtonProps {
   cloudEnvironment: App | ListRuntimeItem;
   permissions: PausePermissionsProvider;
   pauseComputeAndRefresh: (cloudEnvironment: App | ListRuntimeItem) => void;
 }
 
-export function PauseButton(props: PauseButtonProps): ReactNode {
+export const PauseButton = (props: PauseButtonProps): ReactNode => {
   const { cloudEnvironment, permissions, pauseComputeAndRefresh } = props;
   const shouldShowPauseButton =
-    isPauseSupported(getToolLabelFromCloudEnv(cloudEnvironment)) && permissions.canPauseResource(cloudEnvironment);
+    isPauseSupported(getToolLabelFromCloudEnv(cloudEnvironment)) && permissions.hasPausePermission(cloudEnvironment);
 
   return shouldShowPauseButton
     ? h(
         Link,
         {
           style: { marginRight: '1rem' },
-          disabled: !shouldShowPauseButton,
+          disabled: !isComputePausable(cloudEnvironment),
```

Comment:
The disabled doesn't seem to be working quite right, when I tested just now the pause button was enabled for both Paused and Running states. I could hit pause on the paused runtime and got a backend error
<img width="937" alt="image" src="https://github.com/DataBiosphere/terra-ui/assets/85642387/7e5d26fd-dacd-4913-a150-82229f8c8c1a">

Reply:
DMed about this, but posting for clarity in that it seems to be working for me 🤔 
![image](https://github.com/DataBiosphere/terra-ui/assets/6465084/271b26fb-a482-4976-94bb-a8ee9d4fd5eb)

Reply:
this is totally my bad, I had my BEE synched to the wrong `jc-...` branch, it looks now!